### ISSUE TITLE
`--listen` parameter lets one specify nonsensical behaviour

### ISSUE THREAD

**Issue by jeremy-audet** [project-receptor/python-receptor#153]
I can start receptor like so:

```bash
poetry run receptor --data-dir="$(mktemp --directory)" node --listen=receptor://127.0.0.1:8888
```

This works fine. But let's say I start receptor like so:

```bash
poetry run receptor --data-dir="$(mktemp --directory)" node --listen=receptor://127.0.0.1:8888/foo
```

In this case, receptor will (to the best of my knowledge) silently discard the path `/foo`. This seems problematic to me. The user might expect one thing ("receptor will nest its listen interface under the `/foo` path"), but receptor will do something else. This could be a source of user confusion.

If receptor is going to discard the path, then why does it allow a path to be specified in the first place? Perhaps receptor should reject `--listen` arguments with information that'd be dropped anyway. For example, the following URL could be rejected:

```python
>>> from urllib.parse import urlparse
>>> urlparse('receptor://127.0.0.1/0:8888')
ParseResult(scheme='receptor', netloc='127.0.0.1', path='/0:8888', params='', query='', fragment='')
```

The logic could be something like:

```python
if res.path or res.params or res.query or res.fragment:
    raise InvalidListenArgumentError(...)
```

**Comment by jeremy-audet** [comment]
Alternately, receptor could provide parameters like `--listen-scheme`, `--listen-address` and `--listen-port`. That'd be a much more verbose option, though, and it doesn't strike me as being as elegant.

**Comment by ghjm** [comment]
I can get behind the idea of making it an error to provide path information in the URL that won't be used.  That would deal with the receptor://0.0.0.0/0:8888 case.

Does path information possibly mean something in the case of a websocket listener?

**Comment by jeremy-audet** [comment]
> Does path information possibly mean something in the case of a websocket listener?

Good question. I've no idea. :woman_shrugging: @matburt ?

**Comment by jeremy-audet** [comment]
Also worth thinking about: can we tell receptor to bind to an interface, and let it discover that interface' address at runtime? That functionality is both common and useful, and it might affect how this issue is addressed.

**Comment by jeremy-audet** [comment]
Fixed by #157. Automated tests added in #139.
